executor/runtime/docker/docker_linux.go

```diff
@@ -84,6 +84,49 @@ func setupScheduler(cred ucred) error {
 	return nil
 }
 
+func (r *DockerRuntime) movePid1ToUnifiedCgroup(parentCtx context.Context, c runtimeTypes.Container, cred ucred) error {
+
+	cgroupPath := filepath.Join("/proc/", strconv.Itoa(int(cred.pid)), "cgroup")
+	cgroups, err := ioutil.ReadFile(cgroupPath) // nolint: gosec
+	if err != nil {
+		logrus.WithError(err).Error("Could not read container cgroups")
+		return err
+	}
+	for _, line := range strings.Split(string(cgroups), "\n") {
+		cgroupInfo := strings.Split(strings.TrimSpace(line), ":")
+		if len(cgroupInfo) != 3 {
+			continue
+		}
+		controllerType := cgroupInfo[1]
+		if len(controllerType) == 0 {
+			continue
+		}
+		// This is to handle the name=systemd cgroup, we should probably parse /proc/mounts, but this is a little bit easier
+		controllerType = strings.TrimPrefix(controllerType, "name=")
+		if controllerType != "systemd" {
+			continue
+		}
+
+		// systemd needs to be the owner of its systemd cgroup in order to start up
+		controllerPath := cgroupInfo[2]
```

Comment:
Can you make a function called "parseCgroupLine" or something? And it can return controllerType, controllerPath

And it can have some nice unit tests, because takes in a string and returns strings